sigmastate/src/main/scala/org/ergoplatform/ErgoAddress.scala

```diff
@@ -226,24 +230,27 @@ case class ErgoAddressEncoder(networkPrefix: NetworkPrefix) {
           val p = GroupElementSerializer.parse(r)
           new P2PKAddress(ProveDlog(p), contentBytes)
         case Pay2SHAddress.addressTypePrefix =>
+          if (contentBytes.length != 24) { //192-bits hash used
+            throw new Exception(s"Improper content in P2SH script: $addrStr")
+          }
           new Pay2SHAddress(contentBytes)
         case Pay2SAddress.addressTypePrefix =>
           val tree = ErgoTreeSerializer.DefaultSerializer.deserializeErgoTree(contentBytes)
           new Pay2SAddress(tree, contentBytes)
         case _ =>
-          throw new Exception("Unsupported address type: " + addressType)
+          throw new Exception(s"Unsupported address type: $addressType")
       }
     }
   }
 
   object IsPay2SHAddress {
     def unapply(exp: SigmaPropValue): Option[Coll[Byte]] = exp match {
       case SigmaAnd(Seq(
-            BoolToSigmaProp(
-              EQ(
-                Slice(_: CalcHash, ConstantNode(0, SInt), ConstantNode(24, SInt)),
-                ByteArrayConstant(scriptHash))),
-            DeserializeContext(Pay2SHAddress.scriptId, SSigmaProp))) => Some(scriptHash)
+      BoolToSigmaProp(
+      EQ(
+      Slice(_: CalcHash, ConstantNode(0, SInt), ConstantNode(24, SInt)),
+      ByteArrayConstant(scriptHash))),
+      DeserializeContext(Pay2SHAddress.scriptId, SSigmaProp))) => Some(scriptHash)
```

Comment:
The original code is better formatted, more readable.

Reply:
@aslesarenko reverted